isis/src/control/objs/BundleUtilities/BundleObservation.cpp

```diff
@@ -783,6 +783,433 @@ namespace Isis {
+        if (!useDefaultPointing) {
+          m_instrumentRotation->GetPolynomial(coefRA, coefDEC, coefTWI);
+        }
+        // Use the pointing's center angles if not solving for pointing (rotation)
+        else {
+          const std::vector<double> centerAngles = m_instrumentRotation->GetCenterAngles();
+          coefRA[0] = centerAngles[0];
+          coefDEC[0] = centerAngles[1];
+          coefTWI[0] = centerAngles[2];
+        }
+      }
+
+      // for convenience, create vectors of parameters names and values in the correct sequence
+      std::vector<double> finalParameterValues;
+      QStringList parameterNamesList;
+      QStringList correctionUnitList;
+
+        QString str("%1(%2)  ");
+
+        if (nPositionCoefficients > 0) {
+          for (int i = 0; i < nPositionCoefficients; i++) {
+            finalParameterValues.push_back(coefX[i]);
+            if (i == 0) {
+              parameterNamesList.append( str.arg("  X  ").arg("km") );
+              correctionUnitList.append("m");
+            }
+            else {
+              switch(i) {
+              case 1:
+                parameterNamesList.append( str.arg("     ").arg("km/s") );
+                correctionUnitList.append("m/s");
+                break;
+              case 2:
+                parameterNamesList.append( str.arg("     ").arg("km/s^2") );
+                correctionUnitList.append("m/s^2");
+                break;
+              case 3:
+                parameterNamesList.append( str.arg("     ").arg("km/s^3") );
+                correctionUnitList.append("m/s^3");
+                break;
+              default:
+                  parameterNamesList.append(str.arg("     ").arg("km/s^"+toString(i) ) );
+                  correctionUnitList.append("m/s^"+toString(i));
+              }
+            }
+          }
+          for (int i = 0; i < nPositionCoefficients; i++) {
+            finalParameterValues.push_back(coefY[i]);
+            if (i == 0) {
+              parameterNamesList.append( str.arg("  Y  ").arg("km") );
+              correctionUnitList.append("m");
+
+            }
+            else {
+
+              switch(i) {
+
+              case 1:
+                parameterNamesList.append( str.arg("     ").arg("km/s") );
+                correctionUnitList.append("m/s");
+                break;
+              case 2:
+                parameterNamesList.append( str.arg("     ").arg("km/s^2") );
+                correctionUnitList.append("m/s^2");
+                break;
+              case 3:
+                parameterNamesList.append( str.arg("     ").arg("km/s^3") );
+                correctionUnitList.append("m/s^3");
+                break;
+              default:
+                  parameterNamesList.append(str.arg("     ").arg("km/s^"+toString(i) ) );
+                  correctionUnitList.append("m/s^"+toString(i));
+              }
+
+            }
+
+          }
+          for (int i = 0; i < nPositionCoefficients; i++) {
```

Comment:
You loop through nPositionCoefficients three times, but I am not quite sure why? It looks like you can perform all logic in one loop (depending on how your ordering of parameterNamesList and correctionUnitList is meant to be), and you could even handle all three of X, Y and Z in one go (again, depending on ordering).

Reply:
It is to create the proper order in the elements of the list when you later pop them off.  Each image parameter (X,Y,Z,RA,DEC,TWIST) has parameters which correspond to no derivatives (km),first derivative (km/s), second derivative(km/s^2), etc....  (using position as an example).  The number of derivatives of each parameter corresponds to nPositionCoefficients. It is unlikely this value will be greater than 3.  I'm sure there is a better way to do this, but I stuck with the logic of formateBundleOutputString.

Reply:
I believe it was decided to maintain different lists (parameterNamesList and correctionUnitList) for each of the positional arguments (x, y and z) so that we would only need to loop through nPositionCoefficients one time, and then we could combine the lists after the loop?